### PR TITLE
Add a failure test for `test_connectivity`

### PR DESCRIPTION
**Pull request by edthedev**
Add a failure test for `test_connectivity`

### CODE REVIEW

**Comment by edthedev** [comment]
During ensemble this morning we cleared various `lint` and dependency issues.

We left off while MyPy was taking a long time choosing the best `types` versions.
If that process fails, we plan to move from `csoctools02-dev` to our EC2 development environment for the next try.

If that also fails, we could look at the versions in `requirements-test.txt` in `soar-clearpass` and just pin to those.

File: requirements-test.txt [hunk 1]
```diff
@@ -5,18 +5,20 @@ cfgv==3.4.0
...
+exceptiongroup==1.2.2
 executing==2.0.1
 filelock==3.15.4
 flake8==7.1.0
 identify==2.6.0
+idna==3.7
 iniconfig==2.0.0
 ipdb==0.13.13
-ipython==8.26.0
+ipython==8.18.1
```
Comment (mpitcel) [inline requirements-test.txt:16]: The tests indicate that the highest version available is `8.13.0` - what version should we be using here?

Comment (edthedev) [inline requirements-test.txt:16]: Weird. We really don't care. We only use `ipython` when doing interactive debugging.
During the CI/CD build, it doesn't even need to be there.

I'm going to bump it down, and see if the build can continue.

```suggestion
ipython==8.13.0
```

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -13,10 +13,13 @@ on:
       - released
 
 jobs:
-  build:
+  tests:
+    runs-on: ubuntu-20.04
```
Comment (edthedev) [inline .github/workflows/tests.yml:17]: We should bump this forward to the latest LTS release. Our EC2 development box is on Ubuntu 22.04

```suggestion
    runs-on: ubuntu-20.04
```

**Comment by edthedev** [comment]
Bumping the CI/CD runner to a newer version of Ubuntu cleared up a bunch of build issues.

**Review by mpitcel — APPROVED** [review]

**Review by zdc217 — APPROVED** [review]

**Pull request merged**